.travis.yml

```diff
@@ -1,12 +1,15 @@
 language: go
 go:
-  - 1.7
-install: make install-ci
-env:
- # Set higher timeouts for Travis
- - TEST_TIMEOUT_SCALE=20 PACKAGE=github.com/m3db/m3db
+  - 1.8.3
```

Comment:
I definitely think we should but am concerned about overhead. For instance, travis builds are already 6-8min after basic parallelisation, each version we add is going add another 6-8minutes, [example](https://travis-ci.org/m3db/m3db/builds/256186398). Not to mention `dtest` (or more elaborate tests down the road) runs which already take 4-6 hours. 

I'll put both versions in for this PR and write up a 1-pager about what we need to do for M3DB CI improvements.